flake8_trio.py

```diff
@@ -1146,6 +1147,52 @@ def visit_BoolOp(self, node: ast.BoolOp):
         self.uncheckpointed_statements = worst_case_shortcut
 
 
+_startable_targets: Dict[str, Tuple[str, ...]] = {
+    "trio": ("run_process", "serve_ssl_over_tcp", "serve_tcp", "serve_listeners"),
+    "": ("serve",),
+}
+
+
+class Visitor113(Flake8TrioVisitor):
+    def __init__(self, *args: Any, **kwargs: Any):
+        super().__init__(*args, **kwargs)
+        self.async_function = False
+        self.asynccontextmanager = False
+        self.aenter = False
+
+    def visit_AsyncFunctionDef(self, node: ast.AsyncFunctionDef):
+        outer = self.aenter
+        if any(
+            isinstance(n, ast.Name) and n.id == "asynccontextmanager"
+            for n in node.decorator_list
+        ) or (
+            node.name == "__aenter__"
+            and len(node.args.args) == 1
+            and node.args.args[0].arg == "self"
+        ):
+            self.aenter = True
+
+        self.generic_visit(node)
+        self.aenter = outer
+
+    def visit_Yield(self, node: ast.Yield):
+        self.aenter = False
+
+    def visit_Call(self, node: ast.Call) -> None:
+        if (
+            self.aenter
+            # and isinstance(node.func.value.id, ast.Name) and node.func.value.id == nursery ?

```

Comment:
Do you want this check? Could maybe reduce some false alarms, but also miss cases where nurseries aren't named `nursery`

Reply:
Let's stick with the assumption that nurseries are always named nursery for now, false alarms suck and this choice is easier to find and change later.